org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3867,13 +3868,13 @@ class ProgressParams {
 	 * The progress data.
 	 */
 	@NonNull
-	@JsonAdapter(WorkDoneProgressNotificationAdapter.Factory)
-	WorkDoneProgressNotification value
+	@JsonAdapter(ProgressNotificationAdapter)
+	Either<? extends WorkDoneProgressNotification, ? extends JsonElement> value
```

Comment:
I have a concern here that this isn't quite right as this adds Gson into the API where (AFAICT) it does not currently exist. e.g. should this be `Either<? extends WorkDoneProgressNotification, Object>`  which would more closely align it with how `Registration.registerOptions` handles this generic case using `JsonElementTypeAdapter.Factory`?

Reply:
Thank you very much for review, Jonah! I fully agree with your observation and will push a fix in a minute.